### PR TITLE
Cherry-pick #21751 to 7.x: Add proxy metricset

### PR DESCRIPTION
**Pull request by ChrsMark**
Cherry-pick of PR #21751 to 7.x branch. Original message: 

## What does this PR do?
This PR adds `proxy` metricset. It leverages the `use_types` option.

`istiod` (https://github.com/elastic/beats/pull/21519) : light-weight metricset on top of Prometheus to collect metrics from the Prometheus exporter of Istio Daemon
`proxy` : light-weight metricset on top of Prometheus to collect metrics from the Prometheus exporter of each of the proxy sidecars. It is recommended (and documented 🙂 ) to leverage autodiscover feature to enable this metricset so as to dynamically identify sidecars and start monitoring them.

## Why is it important?
To be compatible with versions of Istio higher than `1.5`. 

## Related issues
- Closes https://github.com/elastic/beats/issues/21249
- Relates #15505 https://github.com/elastic/beats/issues/21248

metricbeat-istio-overview
metricbeat-istio-traffic

### CODE REVIEW

File: x-pack/metricbeat/module/istio/proxy/_meta/docs.asciidoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+all these sidecar containers and will start `proxy` metricset for each one of
+them. Here is an example configuration that can be used for that purpose:
+
+
+["source", "yaml"]
+--------------------------------------------
+metricbeat.autodiscover:
+  providers:
+    - type: kubernetes
+      include_annotations: ["prometheus.io.scrape"]
```
Comment (exekias) [inline x-pack/metricbeat/module/istio/proxy/_meta/docs.asciidoc:24]: I see you are not using this at all, is this line needed?

Comment (ChrsMark) [reply x-pack/metricbeat/module/istio/proxy/_meta/docs.asciidoc:24]: nope it's not! I will remove it to avoid for confusion 👍

Comment (exekias) [reply x-pack/metricbeat/module/istio/proxy/_meta/docs.asciidoc]: sorry I saw now this is a backport, we will need to remove it from master too

**Review by exekias — APPROVED** [review]

File: x-pack/metricbeat/module/istio/proxy/_meta/docs.asciidoc (hunk 1 above)
Comment (ChrsMark) [reply x-pack/metricbeat/module/istio/proxy/_meta/docs.asciidoc]: sure thing!

**Pull request merged**